tests/v2/validation/KDM/kdm_test.go

```diff
@@ -0,0 +1,228 @@
+	k.Require().NoError(err, "error getting rancher deployment")
+
+	if deployment.Spec.Replicas == &desiredReplicas {
+		return
+	}
+	deployment.Spec.Replicas = &desiredReplicas
+
+	deployment, err = k.clientset.AppsV1().Deployments(rancherNamespace).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+	k.Require().NoError(err, "error updating rancher deployment")
+
+	// Wait for the deployment to scale up using exponential defaultBackoff
+	err = wait.ExponentialBackoff(defaultBackoff, func() (bool, error) {
+		deployment, err = k.clientset.AppsV1().Deployments(rancherNamespace).Get(context.TODO(), rancherDeployment, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("Error getting deployment: %s", err.Error())
+		}
+
+		if deployment.Status.ReadyReplicas == desiredReplicas {
+			log.Infof("Deployment %s successfully scaled to %d replicas", rancherDeployment, desiredReplicas)
+			return true, nil
+		}
+		log.Infof("Waiting for deployment %s to scale. Current replicas: %d/%d", rancherDeployment, deployment.Status.ReadyReplicas, desiredReplicas)
+		return false, nil
+	})
+	k.Require().NoError(err, "error scaling rancher deployment, timed out")
+
+	// arbitrary sleep for leader election activity to take place on the newly created replicas
+	time.Sleep(15 * time.Second)
+}
+
+func (k *KDMTestSuite) GetRancherReplicas() *v1.PodList {
+	podList, err := k.clientset.CoreV1().Pods(rancherNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: rancherLabelSelector})
+	k.Require().NoError(err, "error getting rancher pod list")
+	return podList
+}
+
+func (k *KDMTestSuite) ExecCMDForKDMDump(pod v1.Pod, cmd []string) string {
+	log.Infof("Exec request for Pod:%v, Command:%v", pod.Name, cmd)
+	request := k.clientset.CoreV1().RESTClient().Get().
+		Resource("pods").
+		Name(pod.Name).
+		Namespace(pod.Namespace).
+		SubResource("exec").
+		SetHeader("Upgrade", "websocket").
+		SetHeader("Sec-Websocket-Key", "websocket").
+		SetHeader("Sec-Websocket-Version", "13").
+		SetHeader("Connection", "Upgrade").
+		VersionedParams(&v1.PodExecOptions{
+			Container: "rancher",
+			Command:   cmd,
+			Stdout:    true,
+			Stderr:    true,
+		}, scheme.ParameterCodec).Do(context.TODO())
+
+	result, err := request.Raw()
+	k.Require().NoError(err, "error executing command")
+	return string(result)
+}
+
+func findLatestVersion(allVersions []string) (string, error) {
+	var latest *semver.Version
+	for _, version := range allVersions {
+		ver, err := semver.NewVersion(version)
+		if err != nil {
+			return "", err
+		}
+		if latest == nil || ver.GreaterThan(latest) {
+			latest = ver
+		}
+	}
+	if latest == nil {
+		return "", fmt.Errorf("no valid versions found")
+	}
+	return latest.String(), nil
+}
+
+func (k *KDMTestSuite) TestChangeKDMurl() {
+	// change kdm url to release
+	k.updateKDMurl("https://raw.githubusercontent.com/vatsalparekh/rancher/kdm-test-dummy/data.json")
```

Comment:
this is dummy right now, to validate a version change happens since currently dev and release are in sync already
would change once review